pkg/operator/controller/externaldns/pod.go

```diff
@@ -170,7 +171,58 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 
 	//TODO: Add logic for the CRD source.
 
+	filterArgs, err := b.domainFilters()
+	if err != nil {
+		return err
+	}
+	container.Args = append(container.Args, filterArgs...)
 	container.Args = append(container.Args, args...)
+	return nil
+}
+
+func (b *externalDNSContainerBuilder) domainFilters() ([]string, error) {
+	var (
+		regexDomainIncludeSet bool
+		regexDomainExcludeSet bool
+		args                  []string
+	)
+	for _, d := range b.externalDNS.Spec.Domains {
+		switch d.FilterType {
+		case operatorv1alpha1.FilterTypeInclude:
+			switch d.MatchType {
+			case operatorv1alpha1.DomainMatchTypeExact:
+				if d.Name == nil {
+					return nil, fmt.Errorf("name for domain cannot be empty")
+				}
+				args = append(args, fmt.Sprintf("--domain-filter=%s", *d.Name))
+			case operatorv1alpha1.DomainMatchTypeRegex:
+				if d.Pattern == nil {
+					return nil, fmt.Errorf("pattern for domain cannot be empty")
+				}
+				if regexDomainIncludeSet {
+					return nil, fmt.Errorf("more than one domain pattern cannot be specified for inclusion")
```

Comment:
Good catch! I think that we need to add this into the API comment.

Reply:
Added as part of doc.